File: backend/crawl/app/models/hitter_stats.py
from sqlalchemy import Column, Integer, String, ForeignKey, Float
from sqlalchemy.orm import relationship
from app.models.database import Base


class HitterStats(Base):
    __tablename__ = "hitter_stats"

    id = Column(Integer, primary_key=True, index=True)
    year = Column(String(10))
    player_id = Column(Integer, ForeignKey("player.id"))
    team_id = Column(Integer, ForeignKey("team.id"))
    avg = Column(String(10))
    game_count = Column(Integer)
    pa = Column(Integer)
    ab = Column(Integer)
    r = Column(Integer)
    h = Column(Integer)
    two = Column(Integer)
    three = Column(Integer)
    homerun = Column(Integer)
    tb = Column(Integer)
    rbi = Column(Integer)
    sac = Column(Integer)
    sf = Column(Integer)
    
    player = relationship("Player", back_populates="hitter_stats")
    team = relationship("Team", back_populates="hitter_stats")

File: backend/baseball/src/main/java/com/newbie/baseball/domain/stats/entity/HitterStats.java
package com.newbie.baseball.domain.stats.entity;

import com.newbie.baseball.domain.player.entity.Player;
import com.newbie.baseball.domain.team.entity.Team;
import jakarta.persistence.*;
import lombok.Getter;

@Getter
@Entity
@Table(name = "hitter_stats")
public class HitterStats {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "id")
    private Integer id;

    @Column(name = "year", length = 10)
    private String year;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "player_id")
    private Player player;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "team_id")
    private Team team;

    @Column(name = "avg", length = 10)
    private String avg;

    @Column(name = "game_count")
    private Integer gameCount;

    @Column(name= "pa")
    private Integer pa;

    @Column(name = "ab")
    private Integer ab;

    @Column(name = "r")
    private Integer r;

    @Column(name = "h")
    private Integer h;

    @Column(name = "two")
    private Integer two;

    @Column(name = "three")
    private Integer three;

    @Column(name = "homerun")
    private Integer homerun;

    @Column(name = "tb")
    private Integer tb;

    @Column(name = "rbi")
    private Integer rbi;

    @Column(name = "sac")
    private Integer sac;

    @Column(name = "sf")
    private Integer sf;
}
